R/image_processing.R

```diff
@@ -27,38 +56,31 @@ process_image <- function(args, prompt, system_instructions) {
 
   # Replace {image_size} placeholder
   if (grepl("\\{image_size\\}", prompt_content)) {
-    img <- image_read(args$submission_image)
-    size <- image_info(img)
-    prompt_content <- gsub("\\{image_size\\}", paste(size$width, "by", size$height), prompt_content)
+    if (!is.null(submission_images)) {
+      img <- image_read(submission_images[1])  # Use first image for size
+      size <- image_info(img)
+      prompt_content <- gsub("\\{image_size\\}", paste(size$width, "by", size$height), prompt_content)
+    }
   }
 
   rendered_prompt <- render_prompt_template(
     prompt_content = prompt_content,
     submission = submission_file,
     solution = solution_file,
-    has_submission_image = grepl("\\{submission_image\\}", prompt_content),
+    has_submission_image = !is.null(submission_images),
```

Comment:
This shouldn't be changed. The intent is conveyed in the prompt text; if the user doesn't pass in images, that can be an error that's reported later.